code/modules/spells/spell_types/rightandwrong.dm

```diff
@@ -190,3 +184,50 @@ GLOBAL_VAR_INIT(summon_magic_triggered, FALSE)
 		log_game("Summon Events was increased!")
 
 #undef SPECIALIST_MAGIC_PROB
+
+/datum/summon_magic_controller
+	var/survivor_probability = 0
+
+/datum/summon_magic_controller/New(survivor_probability)
+	. = ..()
+	src.survivor_probability = survivor_probability
+	RegisterSignal(SSdcs, COMSIG_GLOB_CREWMEMBER_JOINED, .proc/magic_up_new_crew)
+
+	for(var/mob/living/carbon/human/unarmed_human in GLOB.player_list)
+		var/turf/T = get_turf(unarmed_human)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names